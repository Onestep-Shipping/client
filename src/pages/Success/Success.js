import React from 'react';
import './Success.css';
import completedIcon from '../../assets/success-icon.gif';
import {useHistory, useParams} from 'react-router-dom';

import Header from '../../components/Header/Header.js';

const Success = () => {
  const history = useHistory();
  const { type } = useParams();

  let message = "Successfully ";
  let waitingFor = "";

  if (type.startsWith("booking")) {
    message += "booked!";
    waitingFor = "booking confirmation";
  } else if (type.startsWith("bol")) {
    message += "sent!";
    waitingFor = "bill of lading";
  }

  return (
    <div className="homepage-container">
      <Header />
      <div className="body-container2">
        <img className="gif" src={completedIcon} alt="Success!" />
        <text className="success-text">{message}</text>
        <text className="success-text-small">
          We will get back to you as soon as
          possible with {waitingFor} â˜º
        </text>
        <div>
          <button
            className="result-button"
            id="left-button"
            onClick={() => history.push('/schedule')}
          >
            Book More
          </button>
          <button
            className="result-button"
            onClick={() => history.push('/profile')}
          >
            Track Process
          </button>
        </div>
      </div>
    </div>
  );
};

export default Success;
