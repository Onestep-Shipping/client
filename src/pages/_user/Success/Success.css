.success-text {
  font-size: 35px;
  color: rgb(19, 161, 0);
  margin-bottom: 20px;
}

.success-text-small {
  font-size: 16px;
  color: rgb(109, 109, 109);
  margin-bottom: 50px;
}

#left-button {
  margin-right: 30px;
}

.gif {
  margin-top: 20px;
}