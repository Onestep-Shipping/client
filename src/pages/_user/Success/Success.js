import './Success.css';

import {useHistory, useParams} from 'react-router-dom';

import Header from '../../../components/Header/Header.js';
import React from 'react';
import completedIcon from '../../../assets/success-icon.gif';

const Success = () => {
  const history = useHistory();
  const { type } = useParams();

  return (
    <div className="homepage-container">
      <Header />
      <div className="body-container2">
        <img className="gif" src={completedIcon} alt="Success!" />
        <text className="success-text">
          Successfully {type === "booking" ? "booked" : "submitted"}!
        </text>
        <text className="success-text-small">
          We will get back to you as soon as possible with {type === "booking" ? 
          "booking confirmation" : "bill of lading"} â˜º
        </text>
        <div>
          <button
            className="result-button"
            id="left-button"
            onClick={() => history.push('/schedule')}>
            Book More
          </button>
          <button
            className="result-button"
            onClick={() => history.push('/profile')}>
            Track Process
          </button>
        </div>
      </div>
    </div>
  );
};

export default Success;
