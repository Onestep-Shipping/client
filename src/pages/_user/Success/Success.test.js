import React from 'react';
import Success from './Success';
import { shallow } from 'enzyme';

let wrapper = shallow(<Success />);

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), 
  useParams: () => ({
    type: 'booking',
  }),
}));

describe('Success', () => {
  it('should render the Success Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
});