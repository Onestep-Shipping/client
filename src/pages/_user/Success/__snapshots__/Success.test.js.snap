// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success should render the Success Component correctly 1`] = `
<div
  className="homepage-container"
>
  <Header />
  <div
    className="body-container2"
  >
    <img
      alt="Success!"
      className="gif"
      src="success-icon.gif"
    />
    <span
      className="success-text"
    >
      Successfully 
      booked
      !
    </span>
    <span
      className="success-text-small"
    >
      We will get back to you as soon as possible with 
      booking confirmation
       â˜º
    </span>
    <div>
      <button
        className="result-button"
        id="left-button"
        onClick={[Function]}
      >
        Book More
      </button>
      <button
        className="result-button"
        onClick={[Function]}
      >
        Track Process
      </button>
    </div>
  </div>
</div>
`;
